private_gpt/components/ingest/ingest_component.py

```diff
@@ -40,7 +40,12 @@ def __init__(
         self.transformations = transformations
 
     @abc.abstractmethod
-    def ingest(self, file_name: str, file_data: Path) -> list[Document]:
+    def ingest(
+        self,
+        file_name: str,
+        file_data: Path,
+        file_metadata: dict[str, str] | None = None,
```

Comment:
I would say it is better idea to type like `dict[str, Any] | None`

Reply:
This is something I thought about because of llama index's restriction for documents: the metadata can only be a flat value (string, int, float, see https://docs.llamaindex.ai/en/stable/module_guides/loading/documents_and_nodes/usage_documents/#metadata). By using string I made it "fool proof". But lmk if you want Any, i will change it

Reply:
I believe it is better idea try to convert data when we're storing data into vector store that compromise all application with this decision. What do you think @imartinez?